tests/ui/EditTask.test.ts

```diff
@@ -334,32 +334,74 @@ describe('Task editing', () => {
             resetSettings();
         });
 
-        const line = '- [ ] simple';
         it('should change status to Done and add doneDate', async () => {
-            const { waitForClose, container, submit } = await renderTaskModalAndChangeStatus(line, 'x');
+            const { waitForClose, container, submit } = await renderTaskModalAndChangeStatus(
+                '- [ ] expecting done date to be added',
+                'x',
+            );
             expect(getElementValue(container, 'done')).toEqual(today);
 
             submit.click();
-            expect(await waitForClose).toMatchInlineSnapshot('"- [x] simple ✅ 2024-02-29"');
+            expect(await waitForClose).toMatchInlineSnapshot('"- [x] expecting done date to be added ✅ 2024-02-29"');
+        });
+
+        it('should change status to Done and change doneDate', async () => {
+            const { waitForClose, container, submit } = await renderTaskModalAndChangeStatus(
+                '- [ ] expecting done date to be kept ✅ 2024-09-19',
```

Comment:
I don't think this is a very representative example - a task which is TODO but has a DONE date.

The bug report refers to opening the modal, adding a DONE date, and then changing the status DONE...

This test may faithfully represent the current behaviour - but I can imagine changes to the implementation of the modal in future that mean this test still passes, but the bug is reintroduced.

To put it differently, there is a difference between:

- the modal retaining the Done date on entry
- the modal retaining a Done date that the user has added whilst the modal is open.

Reply:
Please take a look at the latest commit: https://github.com/obsidian-tasks-group/obsidian-tasks/pull/3087/commits/5d7b558babb0d49882bc1a0cddd71ce403574d57

I think this test represent the behaviour of the bug. If you agree I will add a couple of similar tests =)

Reply:
Yes it's good. Thanks.

In the new function I su suggest enaming newStatus to newStatusSymbol, for clarity.

Reply:
Refactored to `it.each()` and added another test case (where I discovered the #3089)

Reply:
> Refactored to `it.each()` and added another test case (where I discovered the #3089)

I've told SonarCloud to ignore the TODO.

Resolving conversation.